src/scene/morph-instance.js

```diff
@@ -65,13 +147,160 @@ Object.assign(MorphInstance.prototype, {
         this._dirty = true;
     },
 
+    // generates fragment shader to blend number of textures using specified weights
+    _getFragmentShader: function (numTextures) {
+
+        var i, fragmentShader = '';
+
+        if (numTextures > 0) {
+            fragmentShader += 'varying vec2 uv0;\n' +
+                'uniform highp float morphFactor[' + numTextures + '];\n';
+        }
+
+        for (i = 0; i < numTextures; i++) {
+            fragmentShader += 'uniform highp sampler2D morphBlendTex' + i + ';\n';
+        }
+
+        fragmentShader += 'void main (void) {\n' +
+            '    highp vec4 color = vec4(0, 0, 0, 1);\n';
+
+        for (i = 0; i < numTextures; i++) {
+            fragmentShader += '    color.xyz += morphFactor[' + i + '] * texture2D(morphBlendTex' + i + ', uv0).xyz;\n';
+        }
+
+        fragmentShader += '    gl_FragColor = color;\n' +
+            '}\n';
+
+        return fragmentShader;
+    },
+
+    // creates complete shader for texture based morphing
+    _getShader: function (count) {
+
+        var shader = this.shaderCache[count];
+
+        // if shader is not in cache, generate one
+        if (!shader) {
+            var fs = this._getFragmentShader(count);
+            shader = pc.shaderChunks.createShaderFromCode(this.device, textureMorphVertexShader, fs, "textureMorph" + count);
+            this.shaderCache[count] = shader;
+        }
+
+        return shader;
+    },
+
+    _updateTextureRenderTarget: function (renderTarget, srcTextureName) {
+
+        var device = this.device;
+
+        // blend curently set up textures to render target
+        var submitBatch = function (usedCount, blending) {
+
+            // factors
+            this.morphFactor.setValue(this._shaderMorphWeights);
+
+            // alpha blending - first pass gets none, following passes are additive
+            device.setBlending(blending);
+            if (blending) {
+                device.setBlendFunction(pc.BLENDMODE_ONE, pc.BLENDMODE_ONE);
+                device.setBlendEquation(pc.BLENDEQUATION_ADD);
```

Comment:
You can't use `pc.` in the internal engine code any more.

Reply:
I'll remove it, but I've just tested it and it works.

Reply:
Maybe 'can't' should be 'shouldn't'. :smile:

Reply:
done